src/Microsoft.AspNet.Mvc.Core/RemoteAttribute.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.AspNet.Mvc.Internal;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Routing;
+
+namespace Microsoft.AspNet.Mvc
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public class RemoteAttribute : ValidationAttribute, IClientModelValidator
+    {
+        private string _additionalFields = string.Empty;
+        private string[] _additionalFieldsSplit = new string[0];
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteAttribute"/> class.
+        /// </summary>
+        /// <remarks>
+        /// Intended for subclasses that support URL generation with no route, action, or controller names.
+        /// </remarks>
+        protected RemoteAttribute()
+            : base(Resources.RemoteAttribute_RemoteValidationFailed)
+        {
+            RouteData = new RouteValueDictionary();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteAttribute"/> class.
+        /// </summary>
+        /// <param name="routeName">
+        /// The route name used when generating the URL where client should send a validation request.
+        /// </param>
+        /// <remarks>
+        /// Finds the <paramref name="routeName"/> in any area of the application.
+        /// </remarks>
+        public RemoteAttribute(string routeName)
+            : this()
+        {
+            if (string.IsNullOrWhiteSpace(routeName))
```

Comment:
chose to align with `action == null` meaning "ignore this argument" in `UrlHelper`.  only exception is `areaName` parameter to third constructor; that adds an entry to `RouteData` because the root area has a `null` or empty name.